abstractions/typescript/src/requestAdapter.ts

```diff
@@ -14,64 +14,74 @@ export interface RequestAdapter {
-    responseHandler: ResponseHandler | undefined
+    responseHandler: ResponseHandler | undefined,
+    errorMappings: Record<string, new () => Parsable> | undefined
   ): Promise<ModelType[]>;
   /**
    * Excutes the HTTP request specified by the given RequestInformation and returns the deserialized response model collection.
    * @param requestInfo the request info to execute.
    * @param responseType the class of the response model to deserialize the response into.
    * @param responseHandler The response handler to use for the HTTP request instead of the default handler.
+   * @param errorMappings the error factories mapping to use in case of a failed request.
    * @param type the class of the response model to deserialize the response into.
    * @typeParam ResponseType the type of the response model to deserialize the response into.
    * @return a {@link Promise} with the deserialized response model collection.
    */
   sendCollectionOfPrimitiveAsync<ResponseType>(
     requestInfo: RequestInformation,
     responseType: "string" | "number" | "boolean" | "Date",
-    responseHandler: ResponseHandler | undefined
+    responseHandler: ResponseHandler | undefined,
+    errorMappings: Record<string, new () => Parsable> | undefined
   ): Promise<ResponseType[] | undefined>;
   /**
    * Excutes the HTTP request specified by the given RequestInformation and returns the deserialized primitive response model.
    * @param requestInfo the request info to execute.
    * @param responseHandler The response handler to use for the HTTP request instead of the default handler.
+   * @param errorMappings the error factories mapping to use in case of a failed request.
    * @param responseType the class of the response model to deserialize the response into.
    * @typeParam ResponseType the type of the response model to deserialize the response into.
    * @return a {@link Promise} with the deserialized primitive response model.
    */
   sendPrimitiveAsync<ResponseType>(
     requestInfo: RequestInformation,
     responseType: "string" | "number" | "boolean" | "Date" | "ArrayBuffer",
-    responseHandler: ResponseHandler | undefined
+    responseHandler: ResponseHandler | undefined,
+    errorMappings: Record<string, new () => Parsable> | undefined
   ): Promise<ResponseType>;
   /**
    * Excutes the HTTP request specified by the given RequestInformation and returns the deserialized primitive response model.
    * @param requestInfo the request info to execute.
    * @param responseHandler The response handler to use for the HTTP request instead of the default handler.
+   * @param errorMappings the error factories mapping to use in case of a failed request.
    * @return a {@link Promise} of void.
    */
   sendNoResponseContentAsync(
     requestInfo: RequestInformation,
-    responseHandler: ResponseHandler | undefined
+    responseHandler: ResponseHandler | undefined,
+    errorMappings: Record<string, new () => Parsable> | undefined
```

Comment:
Can you provide an example of how the error mapping  will be used?

Reply:
https://github.com/microsoft/kiota-samples/pull/498/files#diff-962a48495eb64c38f14f62249e24b05451799371dc13b6fc0cc5acd20d423031R90

The request executor method passes in the error mappings to the request adapter so it can use it during deserialization